### PR TITLE
Fixing issues encountered while trying to to-onnx applecider models.

### PR DESCRIPTION
**Pull request by drewoldag**
Fixing assorted issues:

* Found that sometimes the Tensors returned from a model still have auto-grad. enabled - which will prevent converting it back to numpy. So need to `.detach()` in model_exporters.py.
* On mac, calling `torch.default_convert` will try to put tensors on CPU instead of MPS by default. So we try to define the default location for new tensors. before calling default_convert.
* Spelled InferenceDatasetWriter incorrectly when importing
* Output format of ONNX model was different than expected, fixed bug when write it out with InferenceDatasetWriter.
* Copy `to_tensor.py` from train output directory to the ONNX output directory.

### CODE REVIEW

File: src/hyrax/pytorch_ignite.py [hunk 1]
```diff
@@ -517,6 +517,7 @@ def _inner_loop(func, to_tensor, device, config, engine, batch):
     # Note - The `_inner_loop` function is called during the `train` and `infer`
     # verbs when the model is a torch model. Thus we _always_ want the batch to
     # be Tensors.
+    torch.set_default_device(idist.device().type)
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/hyrax/pytorch_ignite.py]: Fixed. Moved this to the `create_engine` function.

**Review by dougbrn — APPROVED** [review]
Looks reasonable to me, just one question!

File: src/hyrax/verbs/engine.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def run(self, model_directory: str = None):
         to_tensor_fn = load_to_tensor(input_directory)
 
         # ~ Load the ONNX model from the input directory.
-        onnx_file_name = input_directory / "model.onnx"
+        onnx_file_name = input_directory / "example_model_opset_20.onnx"
```
[the remark this answers is not in the document]
Comment (drewoldag) [reply src/hyrax/verbs/engine.py]: The more specific file name came from the concatenation of 1) the trained model weights file name and 2) the "operator set" version used when converting to ONNX. 

The opset is roughly equivalent to the ONNX runtime version being used when converting the model. Newer versions include mathematical functions that older versions hadn't yet implemented.
